src/app/modules/customer-management/components/projects/components/project-list/project-list.component.ts

```diff
@@ -47,9 +69,22 @@ export class ProjectListComponent implements OnInit, OnDestroy {
     this.showModal = false;
   }
 
-  openModal(item: Project) {
+  openModal(item: ProjectUI) {
     this.idToDelete = item.id;
-    this.message = `Are you sure you want to delete ${item.name}?`;
+    this.message = `Are you sure you want archive ${item.name}?`;
     this.showModal = true;
   }
+
+  switchStatus(item: ProjectUI): void {
+    if (item.key !== 'inactive') {
+      this.openModal(item);
+    } else {
+      this.showModal = false;
+      this.store.dispatch(new UnarchiveProject(item.id));
+    }
+  }
+
+  setActive(status: any): string {
```

Comment:
I think this is a redundant method.
the status variable  can return `'inactive'` | `'active'` or only `'inactive'`?

Reply:
In some cases the endpoint return `active or null` for active items so there I make sure just return `active or inactive`

Reply:
Ohh, okay. I understand it !